models/classes/TestSessionState/TestSessionStateRestorationService.php

```diff
@@ -0,0 +1,164 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2022 (original work) Open Assessment Technologies SA
+ *
+ */
+
+declare(strict_types=1);
+
+namespace oat\taoQtiTest\models\TestSessionState;
+
+use League\Flysystem\FileNotFoundException;
+use oat\tao\model\state\StateMigration;
+use oat\taoDelivery\model\execution\DeliveryExecution;
+use oat\taoQtiTest\models\QtiTestUtils;
+use oat\taoQtiTest\models\runner\ExtendedState;
+use oat\taoQtiTest\models\runner\time\QtiTimeStorage;
+use oat\taoQtiTest\models\TestSessionService;
+use oat\taoQtiTest\models\TestSessionState\Api\TestSessionStateRestorationServiceInterface;
+use oat\taoQtiTest\models\TestSessionState\Exception\RestorationImpossibleException;
+use Psr\Log\LoggerInterface;
+use qtism\data\AssessmentSection;
+use qtism\data\SectionPartCollection;
+use qtism\data\TestPart;
+use qtism\data\TestPartCollection;
+
+class TestSessionStateRestorationService implements TestSessionStateRestorationServiceInterface
+{
+    /** @var TestSessionService */
+    private $testSessionService;
+    /** @var QtiTestUtils */
+    private $qtiTestUtils;
+    /** @var StateMigration */
+    private $stateMigration;
+    /** @var LoggerInterface */
+    private $logger;
+
+    public function __construct(
+        TestSessionService $testSessionService,
+        QtiTestUtils $qtiTestUtils,
+        StateMigration $stateMigration,
+        LoggerInterface $logger
+    ) {
+        $this->testSessionService = $testSessionService;
+        $this->qtiTestUtils = $qtiTestUtils;
+        $this->stateMigration = $stateMigration;
+        $this->logger = $logger;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function restore(DeliveryExecution $deliveryExecution): void
+    {
+        $deliveryExecutionId = $deliveryExecution->getIdentifier();
+        $userId = $deliveryExecution->getUserIdentifier();
+
+        try {
+            $this->stateMigration->restore($userId, $deliveryExecutionId);
+        } catch (FileNotFoundException $e) {
+            throw new RestorationImpossibleException();
+        }
+
+        $this->restoreExtendedState($userId, $deliveryExecutionId);
```

Comment:
```suggestion
        $this->restoreState($userId, $deliveryExecutionId);
```
WDYT to create One proxy method to contain these 3 calls inside it will help in the future when we need to restore state not forgot to use this 3 calls

Reply:
i don't understand how we can forget call them as they are private methods?